Added data and parameters usage example (#610)
vapor/docs: 4.0/docs/leaf/custom-tags.md

```leaf
The time is #now()
```

## Context Properties

The `LeafContext` contains two important properties. `parameters` and `data` that has everything we should need.

- `parameters`: An array that contains the parameters of the tag.

> - `data`: A dictionary that contains the data of the view.

Reviewer: Nit - clarify
```suggestion
- `data`: A dictionary that contains the data of the view passed to `render(_:_:)` as the context.
```

### Example Hello Tag

> For usage demostration let's implement a simple hello tag using both properties.

Reviewer: Spelling and tidy up
```suggestion
ToDo see how to use this, let's implement a simple hello tag using both properties.
```
